frontend/app/utils/lookup-utils.server.ts

```diff
@@ -27,3 +27,34 @@ export function localizeAndSortRegions(regions: Region[], locale: string) {
   const sortedRegions = mappedRegions.toSorted((a, b) => a.name.localeCompare(b.name, locale));
   return sortedRegions;
 }
+
+/**
+ * Localizes a single language object by adding a localized name.
+ *
+ * @param language - The language object to localize.
+ * @param locale - The locale code for localization.
+ * @returns The localized language object with a localized name.
+ */
+export function localizeLanguage(language: Language, locale: string) {
+  const { nameEn, nameFr, ...rest } = language;
+  return {
+    ...rest,
+    name: (locale === 'fr' ? nameFr : nameEn) ?? '',
+  };
+}
+
+/**
+ * Localizes an array of language objects by adding localized names and sorting them.
+ *
+ * @param language - The array of language objects to localize.
+ * @param locale - The locale code for localization.
+ * @param firstLanguageId - The language ID that specifies the language object that should appear first in the sorted array.
+ * @returns The localized and sorted array of language objects.
+ */
+export function localizeAndSortPreferredLanguages(languages: Language[], locale: string, firstLanguageId: number) {
+  const mappedLanguages = languages.map((language) => localizeLanguage(language, locale));
+  const firstLanguage = mappedLanguages.find((language) => language.id === firstLanguageId.toString());
+  const filteredLanguages = mappedLanguages.filter((language) => language.id !== firstLanguageId.toString());
+  const sortedLanguages = filteredLanguages.toSorted((a, b) => a.name.localeCompare(b.name, locale));
+  return firstLanguage ? [firstLanguage, ...sortedLanguages] : sortedLanguages;
```

Comment:
Can be improved as suggested by @gregory-j-baker

```suggestion
  return mappedLanguages.toSorted((a, b) => {
    if (firstLanguageId && a.id === firstLanguageId.toString()) return -1;
    if (firstLanguageId && b.id === firstLanguageId.toString()) return 1;
    return a.name.localeCompare(b.name, locale);
  });
```

Reply:
Set the `firstLanguageId` param optional.

```
firstLanguageId?: number
```